File: src/java/foundgine-core/src/main/java/com/foundgine/core/execution/mutation/MutationBatchResult.java
package com.foundgine.core.execution.mutation;

import java.util.List;

/** Port of {@code Foundgine.Core.Execution.Mutation.MutationBatchResult}: results from an atomic mutation batch, in execution order. */
public record MutationBatchResult(List<MutationResult> results) {

    public int totalAffectedRows() {
        return results.stream().mapToInt(MutationResult::affectedRows).sum();
    }
}

File: src/csharp/Foundgine.Core/Execution/Mutation/MutationBatchResult.cs
namespace Foundgine.Core.Execution.Mutation;

/// <summary>
/// Results from an atomic mutation batch, in execution order.
/// </summary>
public sealed record MutationBatchResult(
    IReadOnlyList<MutationResult> Results)
{
    public int TotalAffectedRows => Results.Sum(x => x.AffectedRows);
}